File: 1. Two Sum/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println("Hello, World!")
}

func twoSum(nums []int, target int) []int {
	for index, num := range nums {
		for index2, num2 := range nums {
			if index != index2 && num+num2 == target {
				return []int{index, index2}
			}
		}
	}
	return []int{}
}

File: 1. Two Sum/main.cpp
class Solution {
   public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> value_to_index_map;
        int remain;
        for (int i = 0; i < nums.size(); i++) {
            remain = target - nums[i];
            if (value_to_index_map.count(remain)) {
                return vector<int>({i, value_to_index_map[remain]});
            }
            value_to_index_map[nums[i]] = i;
        }
        return vector<int>({0, 0});
    }
};